### PR TITLE
[ENG-2391] Add registration list tests

### PR DESCRIPTION
**Pull request by adlius**
- Ticket: [ENG-2391]
- Feature flag: n/a

## QA Notes

[ENG-2391]: https://openscience.atlassian.net/browse/ENG-2391

### CODE REVIEW

File: tests/integration/components/registries/registration-list/component-test.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    setupIntl(hooks);
+
+    hooks.beforeEach(function(this: RegistrationListTestContext) {
+        this.owner.register('service:router', OsfLinkRouterStub);
+        this.store = this.owner.lookup('service:store');
+        this.intl = this.owner.lookup('service:intl');
+    });
+
+    // tslint:disable-next-line: only-arrow-functions
+    test('list renders and paginates', async function(assert) {
```
Comment (fabmiz) [inline tests/integration/components/registries/registration-list/component-test.ts:28]:
```suggestion
    test('list renders and paginates', async function(this: TestContext, assert) {
```

File: tests/integration/components/registries/registration-list/component-test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import Intl from 'ember-intl/services/intl';
+import { setupIntl } from 'ember-intl/test-support';
+import { setupRenderingTest } from 'ember-qunit';
+import { TestContext } from 'ember-test-helpers';
+import { OsfLinkRouterStub } from '../../../helpers/osf-link-router-stub';
+
+interface RegistrationListTestContext extends TestContext {
+    intl: Intl;
+}
```
Comment (fabmiz) [inline tests/integration/components/registries/registration-list/component-test.ts:14]: `import { TestContext } from 'ember-intl/test-support'`, adds `intl` to `ember-test-helpers`'s `TestContext`
```suggestion
```

File: tests/integration/components/registries/registration-list/component-test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+module('Integration | Component | registration-list', hooks => {
+    setupRenderingTest(hooks);
+    setupMirage(hooks);
+    setupIntl(hooks);
+
+    hooks.beforeEach(function(this: RegistrationListTestContext) {
+        this.owner.register('service:router', OsfLinkRouterStub);
+        this.store = this.owner.lookup('service:store');
+        this.intl = this.owner.lookup('service:intl');
```
Comment (fabmiz) [inline tests/integration/components/registries/registration-list/component-test.ts:24]: `setupIntl(hooks);` takes care of this
```suggestion
```

File: tests/integration/components/registries/registration-list/component-test.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                as |manager|
+            >
+                <Registries::RegistrationList::List @manager={{manager}} />
+            </Registries::RegistrationList::Manager>`);
+        assert.dom('[data-test-registration-list-card]').exists({ count: 10 }, 'First page has 10 elements');
+        await click('[data-test-next-page-button]');
+        assert.dom('[data-test-registration-list-card]').exists({ count: 3 }, 'Second page has 3 elements');
+    });
+
+    test('list renders when no registration is available', async function(this: RegistrationListTestContext, assert) {
```
Comment (fabmiz) [inline tests/integration/components/registries/registration-list/component-test.ts:46]:
```suggestion
    test('list renders when no registration is available', async function(this: TestContext, assert) {
```

**Review by fabmiz — CHANGES_REQUESTED** [review]

**Comment by fabmiz** [comment]
@adlius lint errors

**Comment by fabmiz** [comment]
@adlius conflicts 🤷‍♂️

**Review by futa-ikeda — DISMISSED** [review]

**Comment by fabmiz** [comment]
@adlius could you rebase pls?

**Review by futa-ikeda — APPROVED** [review]

**Review by fabmiz — APPROVED** [review]

**Pull request merged**